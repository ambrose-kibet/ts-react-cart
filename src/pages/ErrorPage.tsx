import { Link } from 'react-router-dom';
const ErrorPage = () => {
  return (
    <div>
      <h1>Error 404 </h1>
      <h2>page Not found</h2>
      <Link to="/">Back Home</Link>
    </div>
  );
};
export default ErrorPage;
