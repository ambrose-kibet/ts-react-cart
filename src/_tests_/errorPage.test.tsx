import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import ErrorPage from '../pages/ErrorPage';
import { describe, it, expect } from 'vitest';

describe('error page', () => {
  it('renders Errorpage correctly', () => {
    const { container } = render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
  it('should  have back home link', () => {
    render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    );
    const backbtn: HTMLAnchorElement = screen.getByRole('link');
    expect(backbtn.href).toContain('/');
  });
});
