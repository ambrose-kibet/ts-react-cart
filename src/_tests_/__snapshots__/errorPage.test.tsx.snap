// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`error page > renders Errorpage correctly 1`] = `
<div>
  <div>
    <h1>
      Error 404 
    </h1>
    <h2>
      page Not found
    </h2>
    <a
      href="/"
    >
      Back Home
    </a>
  </div>
</div>
`;
